### PR TITLE
feature: multiple address support for ledger nano

### PR DESCRIPTION
**Pull request by comountainclimber**
**What current issue(s) from Trello/Github does this address?**
There are no issues open regarding supporting multiple addresses on ledger however it is a feature request I have seen in discord/slack multiple times and pretty neat 🤓 

multiple-ledger-address-support

**What problem does this PR solve?**
This PR adds support for generating public keys from a ledger device.

**How did you solve this problem?**
I expanded the API of the existing neonLedger class.

**How did you make sure your solution works?**
Manually testing of multiple sends and gas claim with a few addresses.

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**
~Those double loading indicators are not my favorite if anyone has any alternative design ideas please comment~
better-loading-indicator https://github.com/CityOfZion/neon-wallet/pull/1750/commits/bc397ef361397c05ec47254b8d5ad85de715ba1b

- [ ] Unit tests written?

### CODE REVIEW

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 1]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+    }))
+  }
+
+  unencodedHexToAddress = (hexString: string) => {
+    const encodedKey = wallet.getPublicKeyEncoded(hexString)
+    return new wallet.Account(encodedKey).address
+  }
+
+  createOptionsFromKeys = () => {
+    const options = this.state.publicKeys.map(key => ({
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: I think `key.key` is a bit weird, how about if you call it `publicKey`, since you are mapping `publicKeys`?

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 2]
```diff
@@ -85,10 +121,26 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   render() {
+    const options = this.createOptionsFromKeys()
+    const { loadingPublicKeys } = this.state
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: new line after last const declaration

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 3]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
     )
   }
 
+  async fetchInitialKeys() {
+    const publicKeys = await getPublicKeys()
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: new line after last const declaration

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 4]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+    this.setState({
+      publicKeys,
+      loadingPublicKeys: false,
+      address: {
+        value: publicKeys[0].key,
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: would be nicer if the initial public key was a constant

```
  const initialPublicKey = publicKeys[0]
```

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 5]
```diff
@@ -58,11 +83,22 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+    if (this.props !== LOADED && nextProps.progress === LOADED) {
+      this.fetchInitialKeys()
+    }
   }
 
   computeStateFromProps = (props: Props) => {
     if (props.progress === LOADED) {
-      return { ledgerStage: CONNECTED, isLoading: false }
+      if (this.intervalId) {
+        // $FlowFixMe
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: why the flow fix?

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 6]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+        label: this.unencodedHexToAddress(publicKeys[0].key)
+      }
+    })
+  }
+
+  async fetchAdditionalKeys() {
+    const { publicKeys } = this.state
+    this.setState({ loadingPublicKeys: true })
+    const lastAccountLoaded = publicKeys[publicKeys.length - 1].account
+    const nextBatchOfKeys = await getPublicKeys(lastAccountLoaded)
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: new line after last const

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 7]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+      address: {
+        value: publicKeys[0].key,
+        label: this.unencodedHexToAddress(publicKeys[0].key)
+      }
+    })
+  }
+
+  async fetchAdditionalKeys() {
+    const { publicKeys } = this.state
+    this.setState({ loadingPublicKeys: true })
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: new line after `this.setState`

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 8]
```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+    const lastAccountLoaded = publicKeys[publicKeys.length - 1].account
+    const nextBatchOfKeys = await getPublicKeys(lastAccountLoaded)
+    this.setState(state => ({
+      publicKeys: [...state.publicKeys, ...nextBatchOfKeys],
+      loadingPublicKeys: false
+    }))
+  }
+
+  unencodedHexToAddress = (hexString: string) => {
+    const encodedKey = wallet.getPublicKeyEncoded(hexString)
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: new line after const

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 9]
```diff
@@ -147,8 +235,15 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   handleLogin = () => {
-    this.props.login(this.props.publicKey)
+    if (this.state.publicKeys.length && this.state.address) {
+      const keyData = this.state.publicKeys.find(
+        key => this.state.address && this.state.address.value === key.key
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: same comment as above, perhaps use `{ key }`

File: app/ledger/neonLedger.js [hunk 10]
```diff
@@ -103,6 +103,21 @@ export default class NeonLedger {
     return Promise.resolve()
   }
 
+  async getPublicKeys(
+    acct: number = 0,
+    unencodedPublicKeys: Array<{ account: number, key: string }> = [],
+    batchSize: number = 10
+  ) {
+    const res = await this.send('80040000', BIP44(acct), [VALID_STATUS])
+    const key = await res.toString('hex').substring(0, 130)
```
Comment (evgenyboxer) [inline app/ledger/neonLedger.js]: new line after last const declaration

File: app/actions/authActions.js [hunk 11]
```diff
@@ -77,14 +79,18 @@ export const nep2LoginActions = createActions(
...
     const publicKeyEncoded = wallet.getPublicKeyEncoded(publicKey)
-    const account = new wallet.Account(publicKeyEncoded)
+    const walletAccount = new wallet.Account(publicKeyEncoded)
 
     return {
       publicKey,
-      address: account.address,
-      signingFunction: ledgerNanoSCreateSignatureAsync,
+      address: walletAccount.address,
+      signingFunction: bindArgsFromN(
```
Comment (evgenyboxer) [inline app/actions/authActions.js]: what are you trying to achieve here? I'm a bit confused by the `bindArgsFromN` function

File: app/components/Inputs/Label/Label.jsx [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+import styles from './Label.scss'
+
+type Props = {
+  label: string
+}
+
+export default class Label extends React.Component<Props> {
+  render() {
+    const { label } = this.props
+    return <label className={styles.label}> {label} </label>
```
Comment (evgenyboxer) [inline app/components/Inputs/Label/Label.jsx]: seems there are extra spaces `> { label } <`
also, shouldn't this be a functional component?

File: app/components/Inputs/Label/Label.jsx [hunk 13]
```diff
@@ -0,0 +1,14 @@
+// @flow
+import React from 'react'
+import styles from './Label.scss'
+
+type Props = {
+  label: string
```
Comment (evgenyboxer) [inline app/components/Inputs/Label/Label.jsx]: we should all this to also be a react component

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 14]
```diff
@@ -23,33 +28,53 @@ const { NOT_CONNECTED, OPEN_APP, CONNECTED } = LEDGER_CONNECTION_STAGES
...
 const POLL_FREQUENCY_MS = 1000
 
 export default class LoginLedgerNanoS extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
-
-    this.state = this.computeStateFromProps(props)
+    this.state = {
+      ...this.computeStateFromProps(props),
+      address: null,
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: I think this is a bit confusing.. usually, an `address` is a string that represents the public address.. but here its actually an option. How about renaming it to `addressOption`?

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 15]
```diff
@@ -85,10 +121,26 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+            {loadingPublicKeys && (
+              <RefreshIcon className={styles.ledgerStageRefreshIcon} />
+            )}
+          </div>
+
+          <StyledReactSelect
+            value={this.state.address}
+            onChange={address => this.setState({ address })}
+            options={options}
+            onMenuScrollToBottom={() => this.fetchAdditionalKeys()}
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: if `fetchAdditionalKeys` was an arrow function, this we could just write `onMenuScrollToBottom={this.fetchAdditionalKeys}`

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 16]
```diff
@@ -147,8 +235,15 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   handleLogin = () => {
-    this.props.login(this.props.publicKey)
+    if (this.state.publicKeys.length && this.state.address) {
+      const keyData = this.state.publicKeys.find(
```
Comment (evgenyboxer) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: not sure if you considered this, but would it help if `publicKeys` was a hashmap instead?

File: app/ledger/neonLedger.js [hunk 17]
```diff
@@ -226,7 +244,7 @@ export const getPublicKey = async (acct: number = 0): Promise<string> => {
 export const getDeviceInfo = async () => {
   const ledger = await NeonLedger.init()
   try {
-    return await ledger.getDeviceInfo()
+    return ledger.getDeviceInfo()
```
Comment (evgenyboxer) [inline app/ledger/neonLedger.js]: no need for `await` here?

File: app/actions/authActions.js (hunk 11 above)
Comment (comountainclimber) [reply app/actions/authActions.js]: So I probably should have left a comment here but binding the account (3rd argument only) to the signingFunc is the only way that this whole process works with the way things are currently set up.

File: app/components/Inputs/Label/Label.jsx (hunk 12 above)
Comment (comountainclimber) [reply app/components/Inputs/Label/Label.jsx]: While there are not extra spaces according to our formatting rules i should expand the API of the component to include additional props like you are suggesting thanks @evgenyboxer was an oversight on my part

File: app/ledger/neonLedger.js (hunk 17 above)
Comment (comountainclimber) [reply app/ledger/neonLedger.js]: I cleaned up the other instance of return await (that was in the code base before this PR) as `return await` is basically useless most of the time... If you are already returning a promise there is no point in awaiting its result (unless there is a catch block to catch errors from another Promise). More info can be found here even though we dont have this rule configured currently https://eslint.org/docs/rules/no-return-await

File: app/actions/authActions.js (hunk 11 above)
Comment (comountainclimber) [reply app/actions/authActions.js]: Previously nothing was passed in so the logic always defaulted to `account = 0`

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 16 above)
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: I think using an array is pretty simple and works fine but maybe I am missing something... using a hashmap was not something I considered

**Comment by comountainclimber** [comment]
@evgenyboxer thanks as always for the very thorough review... This branch went thru multiple iterations sorry I pushed it up with a number of issues. Everything should be addressed based on your feedback please take a look and let me know if you have any additional comments or concerns.

File: app/ledger/neonLedger.js (hunk 17 above)
Comment (comountainclimber) [reply app/ledger/neonLedger.js]: Looks like by removing the `return await` from `getPublicKeys` the functionality breaks totally puzzled by that one 😕

File: app/actions/authActions.js (hunk 11 above)
Comment (dvd-schwrtz) [reply app/actions/authActions.js]: the signing function is for the ledger. Basically it never shares the private key. What happens is that when the user is using a hardware device, the transaction is passed to the ledger in a message, when the user clicks the button, it signs the transaction and sends the signed message back to the application, thereby never revealing the private key that was used to sign the transaction. I am not sure why we are binding anything, but just wanted to clarify how that works in case it is helpful. Please let me know if you would like me to look into this code/ binding logic further

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 16 above)
Comment (dvd-schwrtz) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: I don't think we need a hashmap. there should not be a lot of public keys

File: app/actions/authActions.js (hunk 11 above)
Comment (comountainclimber) [reply app/actions/authActions.js]: @dvd-schwrtz like I mentioned before we have to bind the account otherwise the transaction will always default to account 0 (the first account) => https://github.com/CityOfZion/neon-wallet/blob/ff340cc636eca1165c056cddb86031794a1cd129/app/ledger/neonLedger.js#L264

Comment (comountainclimber) [reply app/actions/authActions.js]: I am binding the third argument (account) when the user authenticates to prevent this...

Comment (dvd-schwrtz) [reply app/actions/authActions.js]: cool. lets change bindArgsFromN to bindActiveAccount or something

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 5 above)
Comment (ranbena) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: nit: no need to check that `intervalId` exists since `clearInterval` handles `undefined` value silently.

File: app/components/Inputs/Label/Label.jsx (hunk 12 above)
Comment (ranbena) [inline app/components/Inputs/Label/Label.jsx]: Perhaps better with text as child instead of string property?
```
return <label className={styles.label}>{this.props.children}</label>
```

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 4 above)
Comment (ranbena) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Or perhaps
```
const { key: initialKey } = publicKeys[0]
```

File: app/ledger/neonLedger.js (hunk 17 above)
Comment (ranbena) [reply app/ledger/neonLedger.js]: Perhaps without the `await`, the `finally` executes `ledger.close()` immediately, therefore not allowing `getDeviceInfo()` to happen 🤔

File: app/components/Inputs/Label/Label.jsx [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+    const { label, loading, renderAdditionalContent } = this.props
+
+    const passDownProps = omit(
+      this.props,
+      'label',
+      'loading',
+      'labelClassName',
+      'labelContainerClassName',
+      'renderAdditionalContent'
+    )
```
Comment (ranbena) [inline app/components/Inputs/Label/Label.jsx]: This `omit` method is used across the app but it not ideal - requires any future changes to props to be mirrored and also perhaps needlessly verbose.

Wdyt of this instead?:
```js
const {
  label,
  loading,
  labelClassName,
  labelContainerClassName,
  renderAdditionalContent,
  ...passDownProps // <-- rest of props
} = this.props

const labelClassName = classNames(styles.label, labelClassName)
const labelContainerClassName = classNames(styles.labelContainer, labelContainerClassName)
```
This way props used in this class will automatically be omitted from the nested components.

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 19]
```diff
@@ -79,16 +122,31 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
+    const options = this.createOptionsFromKeys()
+    const { loadingPublicKeys, publicKeys } = this.state
+
     return (
       <div id="loginLedgerNanoS" className={styles.flexContainer}>
         <form>
           {this.renderStatus()}
+          <Label
+            label="public address:"
+            renderAdditionalContent={this.renderAdditionalLabelContent}
```
Comment (ranbena) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: I think it'd be better to swap this prop for `children`:
```jsx
<Label label="public address:">
  {this.renderAdditionalLabelContent}
</Label>
```

File: app/components/Inputs/Label/Label.scss [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: 5px;
+  min-height: 24px;
+
+  svg {
+    animation: spin 2s linear infinite;
+    margin-left: 5px;
+    path {
+      fill: #5c677f;
```
Comment (ranbena) [inline app/components/Inputs/Label/Label.scss]: Swap with `var(--input-label);`

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 21]
```diff
@@ -106,6 +164,74 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
...
   }
 
+  fetchAdditionalKeys = async () => {
+    const { publicKeys } = this.state
+
+    this.setState({ loadingPublicKeys: true, error: null })
+
+    const lastAccountLoaded = publicKeys[publicKeys.length - 1].account
+
+    const nextBatchOfKeys = await getPublicKeys(lastAccountLoaded + 1).catch(
```
Comment (ranbena) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: From this I deduce that `account` is an array index. If so, I think it should be indicated in the key name (e.g `publicKeys[0].accountIdx`).

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 22]
```diff
@@ -23,46 +28,84 @@ const { NOT_CONNECTED, OPEN_APP, CONNECTED } = LEDGER_CONNECTION_STAGES
...
   login: Function,
   connect: Function,
-  publicKey: ?string,
   error: ?string
 }
 
+type SelectOption = {
+  label: string,
+  value: string
+}
```
Comment (ranbena) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Better this be located in `StyledReactSelect.jsx` (used also in `Settings.jsx`)
And if needed to make more specific (e.g `PriceHistoryPanel` and `ContactsPanel`) we can do this:
```jsx
import type { SelectOption } from '../StyledReactSelect'

type Duration = '1m' | '1w' | '1d'

type DurationSelectOption = {
  ...SelectOption, // <-- copies type definition
  value: Duration // <-- overrides specific type
}
```

Wdyt? This can be a separate PR of course.

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 23]
```diff
@@ -23,46 +28,84 @@ const { NOT_CONNECTED, OPEN_APP, CONNECTED } = LEDGER_CONNECTION_STAGES
...
+
 type State = {
   ledgerStage: LedgerConnectionStage,
-  isLoading: boolean
+  isLoading: boolean,
```
Comment (ranbena) [inline app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Btw, we can use trailing commas for cleaner diffs.
https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

File: app/components/Inputs/Label/Label.jsx (hunk 18 above)
Comment (comountainclimber) [reply app/components/Inputs/Label/Label.jsx]: Haha this is a much cleaner solution that then the pattern used throughout the app thanks @ranbena! Will update.

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 21 above)
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Hmm I see what you are saying... `account` is the var name used throughout the other ledger files - I can update all the other instances if you think its better

Comment (ranbena) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Leave it be them. Let's not go crazy 🤟🏻

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 22 above)
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Sure thats a great idea! However instead of exporting it from `StyledReactSelect` I will just add it to the flow-typed file where all of our global types are defined

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 23 above)
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Sure I am not opposed! In a separate PR I will update our prettier end eslint configurations.

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 24]
```diff
@@ -23,46 +28,84 @@ const { NOT_CONNECTED, OPEN_APP, CONNECTED } = LEDGER_CONNECTION_STAGES
...
+      ...this.computeStateFromProps(props),
+      addressOption: null,
+      publicKeys: [],
+      loadingPublicKeys: true,
+      error: null
+    }
   }
 
-  intervalId: ?number
+  intervalId: IntervalID
```
[the remark this answers is not in the document]
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: 🙌

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 22 above)
Comment (ranbena) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: awesome

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx [hunk 25]
```diff
@@ -147,8 +274,17 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   handleLogin = () => {
-    this.props.login(this.props.publicKey)
+    if (this.state.publicKeys.length && this.state.addressOption) {
+      const keyData = this.state.publicKeys.find(
+        publicKey =>
+          this.state.addressOption &&
```
[the remark this answers is not in the document]
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Flow was throwing an error (even though the check was already being performed on 27) will see if I can come up with a cleaner solution.

[the remark this answers is not in the document]
Comment (ranbena) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: Oh that's cause `this.state` might have changed since the check. Flow is a stickler :)
Just put the addressOption in a new variable and you're good to go:
```js
const { addressOption, publicKeys } = this.state
if (publicKeys.length && addressOption) {
  const keyData = publicKeys.find(
    publicKey => addressOption.value === publicKey.key
  )
  if (keyData) {
    this.props.login(keyData)
  }
}
```

File: app/actions/authActions.js (hunk 11 above)
Comment (comountainclimber) [reply app/actions/authActions.js]: The problem with calling it `bindActiveAccount` is that thats not what its doing... `bindArgsFromN` will bind any amount of function args after a specific argument number (N)

File: app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx (hunk 23 above)
Comment (comountainclimber) [reply app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx]: tracking here https://github.com/CityOfZion/neon-wallet/issues/1754

File: app/components/Inputs/Label/Label.jsx (hunk 12 above)
Comment (comountainclimber) [reply app/components/Inputs/Label/Label.jsx]: I decided to keep the text prop and pass children in for the additional content after reviewing the alternative this seemed to make the most sense and the easiest for styling/layout4
